### ISSUE TITLE
feat(sphere): UniformOnSphere + VonMisesFisher distributions and tangent-basis utilities

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/gauss_flows#52]
## Problem / Request

Add a `UniformOnSphere(d)` and `VonMisesFisher(d, mean, concentration)` distribution to `gauss_flows._src.distributions.sphere`, plus a `tangent_basis(x)` utility used by every spherical bijection.

## User Story

> As a user fitting a flow on the sphere, I want a uniform base distribution and a VMF target distribution that interoperate with `flowjax.distributions.Transformed`, so that I can write `Transformed(UniformOnSphere(d=2), my_sphere_flow).log_prob(x)` exactly the same way I would for the Euclidean case.

## Motivation

Every spherical normalizing flow needs a uniform base distribution on S^d. The VMF distribution is broadly useful: it is the canonical target for directional-statistics benchmarks (it is the distribution behind the 4-mode mixture in the Rezende et al. 2020 paper) and a useful prior for NumPyro-based directional-data models. The `tangent_basis` utility is shared infrastructure for every bijection in this wave (used to project the (d+1)×(d+1) Jacobian down to the intrinsic d×d log-det).

## Proposed API

```python
# src/gauss_flows/_src/distributions/sphere.py

class UniformOnSphere(AbstractDistribution):
    """Uniform distribution on S^d ⊂ ℝ^{d+1}.

    Shape:
        Event shape: (d+1,). Sample shape leading dims as usual.
        Samples satisfy ‖x‖₂ = 1 to numerical precision.
        log_prob is constant: -log surface_area(S^d).
    """
    d: int

    def __init__(self, d: int): ...
    def _sample(self, key, sample_shape) -> Array: ...
    def _log_prob(self, x) -> Array: ...


class VonMisesFisher(AbstractDistribution):
    """Von Mises–Fisher distribution on S^d ⊂ ℝ^{d+1}.

    p(x | μ, κ) ∝ exp(κ · μᵀx),    x, μ ∈ S^d, κ ≥ 0.
    """
    mean: Array       # shape (d+1,), unit vector
    concentration: Array  # scalar, ≥ 0

    def __init__(self, mean, concentration): ...
    def _sample(self, key, sample_shape) -> Array: ...    # Wood (1994) rejection sampler
    def _log_prob(self, x) -> Array: ...                  # closed form, normalised


# src/gauss_flows/_src/transforms/_sphere_utils.py

def tangent_basis(x: Float[Array, " d_plus_1"]) -> Float[Array, "d_plus_1 d"]:
    """Orthonormal basis E ∈ ℝ^{(d+1)×d} of the tangent space T_x S^d.

    Used by exp-map flows to compute log|det J| = ½ log det(EᵀJᵀJE).
    Implementation: QR of any matrix whose columns span x⊥.
    """

def expmap_sphere(
    x: Float[Array, " d_plus_1"], v: Float[Array, " d_plus_1"]
) -> Float[Array, " d_plus_1"]:
    """Exp map on S^d at base point x with tangent vector v ∈ T_x S^d.

        expmap(x, v) = x · cos(‖v‖) + (v / ‖v‖) · sin(‖v‖).

    Numerically safe at ‖v‖ → 0 via a small-angle Taylor expansion.
    """

def logmap_sphere(
    x: Float[Array, " d_plus_1"], y: Float[Array, " d_plus_1"]
) -> Float[Array, " d_plus_1"]:
    """Inverse of expmap: tangent vector pointing from x to y on S^d."""
```

## Mathematical Notes

```text
UniformOnSphere log-density:
    log p(x) = -log A_d
    where  A_d = 2 · π^((d+1)/2) / Γ((d+1)/2)   (surface area of S^d in ℝ^{d+1})

Von Mises–Fisher log-density on S^d:
    log p(x | μ, κ)
        = κ · μᵀx
          + (d/2 − 1) · log(κ)
          − (d/2 + 1/2) · log(2π)
          − log I_{d/2 − 1}(κ)
    where I_ν is the modified Bessel function of the first kind.

Tangent basis identity:
    For x ∈ S^d, the tangent space is {v ∈ ℝ^{d+1} : xᵀv = 0}.
    E ∈ ℝ^{(d+1)×d} satisfies  EᵀE = I_d,  E Eᵀ = I_{d+1} − x xᵀ.
    Build via QR(I_{d+1} − x xᵀ) and drop the zero column, OR via
    Householder reflection mapping x to e₁ then taking columns 2..d+1.
```

## Acceptance criteria

- [ ] `UniformOnSphere(d).sample(key, (N,)).shape == (N, d+1)` and `‖samples‖ ≈ 1` to atol=1e-6 for d ∈ {1, 2, 3, 7}
- [ ] `UniformOnSphere(d).log_prob(x)` matches `-log(surface_area(S^d))` exactly
- [ ] Sample mean of N=10⁵ samples ≈ 0 to atol=1e-2 (uniformity check)
- [ ] `VonMisesFisher` log-density matches `scipy.stats.vonmises_fisher` to atol=1e-5 across (d, κ) grid; sampler mean concentrates on `mean` with the right `MRL(κ)` (mean-resultant-length) to atol=1e-2 over N=10⁴
- [ ] `tangent_basis(x)` satisfies `EᵀE = I_d` and `xᵀE = 0` to atol=1e-6, vmappable
- [ ] `expmap_sphere` and `logmap_sphere` roundtrip to atol=1e-5 for `‖v‖ < π`
- [ ] All four are jit/vmap/grad-friendly (autodiff smoke test)
- [ ] Re-exported from `gauss_flows` top-level

## Out of scope

- Inverse VMF sampler beyond Wood (1994) — Wood is sufficient.
- Mixture-of-VMF distribution (composing several VMFs is left to user / NumPyro).
- Spherical-harmonic representations.

## Estimate

~150 LOC implementation + ~150 LOC tests. Small PR.
